Sources/SwiftParser/IncrementalParseTransition.swift

```diff
@@ -46,9 +46,13 @@ public final class IncrementalParseReusedNodeCollector:
 /// Keeps track of a previously parsed syntax tree and the source edits that
 /// occurred since it was created.
 public final class IncrementalParseTransition {
-  fileprivate let previousTree: SourceFileSyntax
-  fileprivate let edits: ConcurrentEdits
-  fileprivate let reusedDelegate: IncrementalParseReusedNodeDelegate?
+  fileprivate var previousTree: SourceFileSyntax?
+  fileprivate var edits: ConcurrentEdits?
+  fileprivate var reusedDelegate: IncrementalParseReusedNodeDelegate?
+
+  public var furthestOffset: Int?
```

Comment:
Split `IncrementalParseTransition` into two types as discussed offline.

Reply:
I add a `IncrementalParseNodeAffectRangeCollector` to collect this and make it as a trigger to enable incremental parse.